From 98b1208e869acd087a622e131e7b11e179ee05d0 Mon Sep 17 00:00:00 2001
From: Frank Wang <frank.wang@rock-chips.com>
Date: Thu, 9 Nov 2017 10:45:24 +0800
Subject: [PATCH] usb: dwc_otg_310: delete legacy cru-soft-rst API

Since the DWC USB2.0 host of Rockchip use dwc2 driver instead of
dwc_otg_310 driver, so delete these legacy hack code which contain
the old cru-soft-rst API directly.

Change-Id: I2990c111481a14084c8aa96a54192b9fe88aa61c
Signed-off-by: Frank Wang <frank.wang@rock-chips.com>
---
 drivers/usb/dwc_otg_310/dwc_otg_cil_intr.c | 30 ----------------------
 1 file changed, 30 deletions(-)

diff --git a/drivers/usb/dwc_otg_310/dwc_otg_cil_intr.c b/drivers/usb/dwc_otg_310/dwc_otg_cil_intr.c
index 0a29f41c0347..c8e590929d65 100644
--- a/drivers/usb/dwc_otg_310/dwc_otg_cil_intr.c
+++ b/drivers/usb/dwc_otg_310/dwc_otg_cil_intr.c
@@ -499,34 +499,6 @@ void w_wakeup_detected(void *data)
 	core_if->lx_state = DWC_OTG_L0;
 }
 
-static inline void rk_dwc_otg_phy_soft_reset(dwc_otg_core_if_t *core_if)
-{
-	struct dwc_otg_platform_data *pldata;
-	guid_data_t guid;
-
-	pldata = core_if->otg_dev->pldata;
-	guid.d32 = core_if->core_global_regs->guid;
-
-	if ((is_rk3288_usb()) && ((guid.d32 & 0x01) == 0)) {
-		/* only used for HOST20, OTG HOST do not need.
-		 * first, do soft reset usb phy, and then usb phy
-		 * can drive resume signal.
-		 * second, set usb phy into suspend mode and
-		 * normal mode once time, and then usb phy can
-		 * send SOFs immediately after resume.
-		 */
-		rk3288_cru_set_soft_reset(RK3288_SOFT_RST_USBHOST1PHY, true);
-		udelay(50);
-		rk3288_cru_set_soft_reset(RK3288_SOFT_RST_USBHOST1PHY, false);
-		udelay(50);
-
-		pldata->phy_suspend(pldata, USB_PHY_SUSPEND);
-		udelay(50);
-		pldata->phy_suspend(pldata, USB_PHY_ENABLED);
-		udelay(100);
-	}
-}
-
 /**
  * This interrupt indicates that the DWC_otg controller has detected a
  * resume or remote wakeup sequence. If the DWC_otg controller is in
@@ -605,8 +577,6 @@ int32_t dwc_otg_handle_wakeup_detected_intr(dwc_otg_core_if_t *core_if)
 			DWC_MODIFY_REG32(core_if->pcgcctl, pcgcctl.d32, 0);
 			udelay(10);
 
-			rk_dwc_otg_phy_soft_reset(core_if);
-
 			DWC_TASK_SCHEDULE(core_if->wkp_tasklet);
 		} else {
 			/** Change to L0 state*/
-- 
2.35.3

